src/Build/Evaluation/Expander.cs

```diff
@@ -775,47 +772,99 @@ private static int ScanForClosingQuote(char quoteChar, string expression, int in
+            // This shouldn't happen
+            int firstSlice = 0;
+            int lastSlice = slices.Count - 1;
+            if (lastSlice == -1)
+            {
+                arguments.Add("");
+                return;
+            }
+            // from end
+            int lastSliceIdx = slices[slices.Count - 1].Item2;
+            // from start
+            int firstSliceIdx = slices[0].Item1;
 
-            // We support passing of null through the argument constant value null
-            if (String.Equals("null", argValue, StringComparison.OrdinalIgnoreCase))
+            // Trim from the start
+            while (firstSlice <= lastSlice && Char.IsWhiteSpace(arg, firstSliceIdx))
             {
-                arguments.Add(null);
+                firstSliceIdx++;
+                if (firstSliceIdx > slices[firstSlice].Item2 && firstSlice < lastSlice)
+                {
+                    firstSlice++;
+                    firstSliceIdx = slices[firstSlice].Item1;
+                }
             }
-            else
+
+            // Trim from the end.
+            // There is some extra logic here to avoid edge case where we would trim a whitespace only character with
+            // one slice from the start, and then once more from the end, resulting in invalid indices.
+            while (((firstSlice < lastSlice) || (firstSlice == lastSlice && firstSliceIdx < lastSliceIdx)) && Char.IsWhiteSpace(arg, lastSliceIdx - 1))
             {
-                if (argValue.Length > 0)
+                lastSliceIdx--;
+                if (slices[lastSlice].Item1 > lastSliceIdx && firstSlice < lastSlice)
                 {
-                    if (argValue[0] == '\'' && argValue[argValue.Length - 1] == '\'')
-                    {
-                        arguments.Add(argValue.Trim(s_singleQuoteChar));
-                    }
-                    else if (argValue[0] == '`' && argValue[argValue.Length - 1] == '`')
-                    {
-                        arguments.Add(argValue.Trim(s_backtickChar));
-                    }
-                    else if (argValue[0] == '"' && argValue[argValue.Length - 1] == '"')
-                    {
-                        arguments.Add(argValue.Trim(s_doubleQuoteChar));
-                    }
-                    else
-                    {
-                        arguments.Add(argValue);
-                    }
+                    lastSlice--;
+                    lastSliceIdx = slices[lastSlice].Item2;
                 }
-                else
+            }
+
+            if (firstSliceIdx == lastSliceIdx)
+            {
+                arguments.Add("");
+                return;
+            }
+
+            bool removedQuotes = false;
+
+            // If the argument is in quotes, we want to remove those
+            if ((arg[firstSliceIdx] == '\'' && arg[lastSliceIdx - 1] == '\'') ||
+                (arg[firstSliceIdx] == '`' && arg[lastSliceIdx - 1] == '`') ||
```

Comment:
The comparison didn't use them (I'm not 100% sure why, but I kept it that way). Since I'm working with indices, I didn't need them for .Trim() calls anymore so they became unused and the ./build.cmd started complaining.